tests/integration2/pilot/sidecar_api_test.go

```diff
@@ -0,0 +1,195 @@
+		func(response *xdsapi.DiscoveryResponse) (b bool, e error) {
+			validator := structpath.AssertThatProto(t, response)
+			if !validator.Accept("{.resources[?(@.address.socketAddress.portValue==%v)]}", mixerCheckPort) {
+				return false, nil
+			}
+			validateListenersNoConfig(t, validator, mixerCheckPort)
+			return true, nil
+		})
+	if err != nil {
+		t.Fatalf("Failed to test as no resource accepted: %v", err)
+	}
+
+	// Apply some config
+	path, err := filepath.Abs("../../testdata/config")
+	if err != nil {
+		t.Fatalf("No such directory: %v", err)
+	}
+	err = gal.ApplyConfigDir(path)
+	if err != nil {
+		t.Fatalf("Error applying directory: %v", err)
+	}
+
+	// Now continue to watch on the same stream
+	err = pilot.WatchDiscovery(time.Second*5000,
+		func(response *xdsapi.DiscoveryResponse) (b bool, e error) {
+			validator := structpath.AssertThatProto(t, response)
+			if !validator.Accept("{.resources[?(@.address.socketAddress.xxxValue==%v)]}", mixerCheckPort) {
+				return false, nil
+			}
+			validateListenersNoConfig(t, validator, mixerCheckPort)
+			return true, nil
+		})
+	if err != nil {
+		t.Fatalf("Failed to test as no resource accepted: %v", err)
+	}
+
+
+}
+
+/*
+	resp, err := pilot.CallDiscovery(req)
+	for i := 0; i < 10; i++ {
+		 log.Debugf("Listener length %v", len(resp.Resources))
+		 time.Sleep(time.Second * 2)
+		  resp, err = pilot.CallDiscovery(req)
+	}
+
+	if err != nil {
+		t.Fatalf("discovery failed: %v", err)
+	}
+*/
+
+func validateListenersNoConfig(t *testing.T, response *structpath.Structpath, mixerCheckPort int) {
+	t.Run("validate-mixer-listener", func(t *testing.T) {
+		mixerListener := response.ForTest(t).
+			Select("{.resources[?(@.address.socketAddress.portValue==%v)]}", mixerCheckPort)
+
+		mixerListener.
+			Equals("0.0.0.0", "{.address.socketAddress.address}").
+			// Example doing a struct comparison, note the pain with oneofs....
+			Equals(&core.SocketAddress{
+				Address: "0.0.0.0",
+				PortSpecifier: &core.SocketAddress_PortValue{
+					PortValue: uint32(mixerCheckPort),
+				},
+			}, "{.address.socketAddress}").
+			Select("{.filterChains[0].filters[0]}").
+			Equals("envoy.http_connection_manager", "{.name}").
+			Equals(true, "{.config.generate_request_id}").
+			Equals("mixer envoy.cors envoy.fault envoy.router", "{.config.http_filters[*].name}").
+			Select("{.config}").
+			Exists("{.rds.config_source.ads}").
+			Exists("{.stat_prefix}").
+			Equals(100, "{.tracing.client_sampling.value}").
+			Equals(100, "{.tracing.overall_sampling.value}").
+			Equals(100, "{.tracing.random_sampling.value}").
+			Equals("EGRESS", "{.tracing.operation_name}").
+			Equals("websocket", "{.upgrade_configs[*].upgrade_type}").
+			Equals(false, "{.use_remote_address}")
+
+		mixerListener.
+			Equals(false, "{.deprecatedV1.bindToPort}").
+			NotExists("{.useOriginalDst}")
+
+		mixerListener.
+			Select("{.filterChains[0].filters[0].config.http_filters[?(@.name==\"mixer\")].config}").
```

Comment:
nit: this looks like it could benefit from table-driven tests (https://github.com/golang/go/wiki/TableDrivenTests)

Reply:
Its generally heading in that direction though its a little tricky because of the need to evaluate ADS events to see if they can be tested or not. See note above about test flow changes.

Will look into this more seriously in follow up